examples/hmcecs.py

```diff
@@ -0,0 +1,147 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: Hamiltonian Monte Carlo with Energy Conserving Subsampling
+=========================
```

Comment:
Nice example! Could you add a png plot to [this folder](https://github.com/pyro-ppl/numpyro/tree/master/docs/source/_static/img/examples) and add this example to the Applications section of this [index file](https://github.com/pyro-ppl/numpyro/blob/master/docs/source/index.rst)?

Reply:
Yes, np, and I'll fix the equals.